### DISCUSSION TITLE
Add option to shift a bird record

### DISCUSSION THREAD

**Discussion by JP-MCA** [mcguirepr89/BirdNET-Pi#582] [Ideas]
I would like to propose a new feature.
Actually, I have a hearing impairment which does not allows my ears to catch anything upper than 6500 Hz.
I was not aware that there where so much birds singing in my garden with frequencies upper than 6500 Hz, but I had the prove by looking at birdnet-PI spectrograms.
What a pitty that I could not enjoy of those beautiful songs :-(

But, after having read the thread about bats detection, I come up with the idea that we could add a little tickbox near each spectrogram, to ask ffmpeg to shift frequencies down, which would be an easy manner to allows to ear some birds songs inaccessible for some of us.

In fact, the process is very easy, because we already have ffmpeg onboard of our RPI.
We just have to modify some Python scripts to insert following command for shifting frequencies down:

```
<pseudo-code>
if tickbox == YES
    ffmpeg -i the_original_recorded_file.mp3 -af "rubberband=pitch=3000/6000" the_shifted_recorded_file.mp3
endif
</pseudo-code>
```

and that's it. In this case, this is a division by 2 in frequency.

**Could someone help me finding the best place to add this new feature in my birdnet-PI instance ?
I could then test it, and possibly send a pull request.**

I think that this tickbox should be present near the PLAY button, so that we could activate it or not, whether we want to shift or not.

**Comment by ehpersonal38** [comment]
Hi @JP-MCA !

I think this is a fantastic idea - what a great way to utilize technology to make birding more accessible. 

On the implementation of it, there are two options here:
1) Implement FFMPEG to pitch down _all_ recordings as they are recorded, when a checkbox is set in Tools -> Settings
2) Implement this on the browser side using Javascript. We'll have to override the default browser audio player since that doesn't actually allow you to pitch mp3's down. A (lot) trickier but it's closer to the implementation you wanted originally, with the tickbox toggle button and all

**Reply by JP-MCA** [reply]
I fear that your option1 would not be acceptable by the classifyer model, and that a lot of bird recognition are going to fail. I mean: the automatic classification depends on frequencies, and if we feed the classifier with shifted recordings, it will fail for sure

Could we implement a new web service that would execute the FFMPEG command on a given MP3 path ?
That way, we would avoid to touch every recordings (option1), or to modify default browser audio player (option2) which can be a problem because there are many different implementation depending on the running platform (Android browser, OSX, Windows, etc).

My idea is that when you click on the little checkbox, a call would be made to "play.php" to execute FFMPEG and thus create a new MP3 on the fly. This would generate a temporary MP3 file, which would then be used by "play.php" to present the shifted MP3 to the user.

**Reply by ehpersonal38** [reply]
I agree now that option 1 wouldn't work.

Okay your idea is certainly possible. The tick button would make an AJAX call to a php file that shell_exec()'s the ffmpeg command and returns the name of the new mp3 file, which then replaces the audio source's src. Were you interested in doing this yourself or if not I can probably have a go at it whenever I have free time

**Reply by croisez** [reply]
I have the capability to make it myself. I can try, yes.
As I am not used to the architecture of this project, as soon as I have it working, I could let you review my dev.

**Reply by croisez** [reply]
> I have the capability to make it myself. I can try, yes. As I am not used to the architecture of this project, as soon as I have it working, I could let you review my dev.

sorry, I made a mistake at the first place, by using JP-MCA account, which is a profile I use for work. Croisez is my personal account.

**Reply by croisez** [reply]
Where could I discuss about technical issues ? I would like to get some clue about the best implementations. Could we discuss when you have time ? Could we use another channel to not ruin this thread ? => Teams, Whatsapp, ... ?

**Reply by ehpersonal38** [reply]
Sure, try joining this [https://gitter.im/BirdNET-Pi/community](https://gitter.im/BirdNET-Pi/community?utm_source=share-link&utm_medium=link&utm_campaign=share-link)
